extensions/auth/portability-auth-microsoft/src/main/java/org/dataportabilityproject/auth/microsoft/MicrosoftAuthServiceExtension.java

```diff
@@ -1,41 +1,78 @@
+/*
+ * Copyright 2018 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package org.dataportabilityproject.auth.microsoft;
 
-import com.google.common.collect.ImmutableList;
-import java.util.List;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.OkHttpClient;
 import org.dataportabilityproject.api.launcher.ExtensionContext;
+import org.dataportabilityproject.spi.cloud.storage.AppCredentialStore;
 import org.dataportabilityproject.spi.gateway.auth.AuthDataGenerator;
-import org.dataportabilityproject.spi.gateway.auth.extension.AuthServiceExtension;
 import org.dataportabilityproject.spi.gateway.auth.AuthServiceProviderRegistry.AuthMode;
+import org.dataportabilityproject.spi.gateway.auth.extension.AuthServiceExtension;
+import org.dataportabilityproject.types.transfer.auth.AppCredentials;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
 
 /** */
 public class MicrosoftAuthServiceExtension implements AuthServiceExtension {
-  private MicrosoftAuthDataGenerator contactsGenerator;
+  private static final String REDIRECT_PATH = "/callback/microsoft";
 
-  public MicrosoftAuthServiceExtension() {
-    // contactsGenerator = new MicrosoftAuthDataGenerator(r)
-  }
+  private List<String> supportedServices = Arrays.asList("calendar", "contacts");
```

Comment:
private static final ImmutableList<String> SUPPORTED_SERVICES ?

Reply:
OK will do

Reply:
On the certs question, they're self-signed so they will throw up a warning on any browser or VM usage since they won't be trusted.